### ISSUE TITLE
Missing label on button that removes files from the drag-and-drop file upload field

### ISSUE THREAD

**Issue by fvwanja** [solspace/craft3-freeform#910]
**Describe the bug or issue you're experiencing**

**Steps to reproduce**

1. Add a field of type „File Drag & Drop”
2. Add files in the frontend
3. Use a screenreader to navigate the page and focus on the „X” button to remove a file / or hover with the mouse on the „X” button

![Screenshot 2023-11-08 at 11 10 11](https://github.com/solspace/craft-freeform/assets/90781427/7c7da6fb-746e-4873-ab07-1d56f08c430b)

**Expected behavior**

1. The button „X” should have an attribute `title` so when hovering with a mouse, we get the information, that this icon will remove a file
2. The button „X” should have an attribute `aria-label` so when focussing it with the keyboard, screenreaders read the correct functionality. Currently they read „link” (tested on Apple VoiceOver).

The values for `title` and `aria-label` should be the same. It could be `"Remove file"`.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.5.9
- Freeform Version: 4.1.11
- Freeform Edition: LITE trial
- Fresh Install or Upgrade: Fresh

**Additional context**

The accessibility overall is okay, could be better, maybe I will create a feature request for that some day.
